src/main/java/com/launchdarkly/client/FeatureFlag.java

```diff
@@ -123,10 +122,18 @@ private Integer evaluateIndex(LDUser user) {
     return fallthrough.variationIndexForUser(user, key, salt);
   }
 
-  private JsonElement getVariation(Integer index) {
-    if (index == null || index >= variations.size()) {
+  private JsonElement getVariation(Integer index) throws EvaluationError {
```

Comment:
We distinguish the fallthrough rule not matching (index is null) from the index not referencing a valid variation, which would indicate that the rules aren't well-formed.